samples/server/petstore/python-flask/openapi_server/test/test_attribute_map_order.py

```diff
@@ -0,0 +1,68 @@
+# coding: utf-8
+
+from __future__ import absolute_import
+import unittest
+from datetime import datetime
+
+from openapi_server.models.api_response import ApiResponse  # noqa: E501
+from openapi_server.models.order import Order
+from openapi_server.test import BaseTestCase
+
+
+class TestOrderAttributeMap(BaseTestCase):
+    """Model serialization has to respect attribute_map on the class"""
+
+    def setUp(self) -> None:
+        now = datetime.now()
+
+        self.data = {
+            "id": 0,
+            "pet_id": 1,
+            "quantity": 25,
+            "ship_date": now,
+            "status": "placed",
+            "complete": True
+        }
+
+        self.desired_data = {
+            "id": 0,
+            "petId": 1,
+            "quantity": 25,
+            "shipDate": now,
+            "status": "placed",
+            "complete": True
+        }
+
+    def test_order_to_dict_default(self):
+        """
+        By default `attribute_map` uses `True` value,
+        attribute_map of the class is respected.
+        """
+        order = Order(**self.data)
+        dikt = order.to_dict()
+        self.assertEqual(dikt, self.desired_data)
+
+    def test_order_to_dict_false(self):
+        """
+        Passing `False` into `to_dict` method does not use
+        `attribute_map` on the class.
+        """
+        order = Order(**self.data)
+        dikt = order.to_dict(attr_map=False)
+        self.assertEqual(dikt, self.data)
+
+    def test_order_from_dict_to_dict(self):
+        """
+        Creates new `Order` instance by using `from_dict` method,
+        passing in data from existing `Order` `to_dict` method.
+        Instances and their `dict` data are equal.
+        """
+        order = Order(**self.data)
+        order2 = Order.from_dict(order.to_dict())
```

Comment:
````python
______________ TestOrderAttributeMap.test_order_from_dict_to_dict ______________

self = <openapi_server.test.test_attribute_map_order.TestOrderAttributeMap testMethod=test_order_from_dict_to_dict>

    def test_order_from_dict_to_dict(self):

        """

        Creates new `Order` instance by using `from_dict` method,

        passing in data from existing `Order` `to_dict` method.

        Instances and their `dict` data are equal.

        """

        order = Order(**self.data)

>       order2 = Order.from_dict(order.to_dict())

openapi_server/test/test_attribute_map_order.py:61: 

_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

openapi_server/models/order.py:70: in from_dict

    return util.deserialize_model(dikt, cls, attr_map=attr_map)

openapi_server/util.py:121: in deserialize_model

    setattr(instance, attr, _deserialize(value, attr_type))

openapi_server/util.py:26: in _deserialize

    return deserialize_datetime(data)

openapi_server/util.py:89: in deserialize_datetime

    return parse(string)

.tox/py3/lib/python3.6/site-packages/dateutil/parser/_parser.py:1374: in parse

    return DEFAULTPARSER.parse(timestr, **kwargs)

.tox/py3/lib/python3.6/site-packages/dateutil/parser/_parser.py:646: in parse

    res, skipped_tokens = self._parse(timestr, **kwargs)

.tox/py3/lib/python3.6/site-packages/dateutil/parser/_parser.py:725: in _parse

    l = _timelex.split(timestr)         # Splits the timestr into tokens

.tox/py3/lib/python3.6/site-packages/dateutil/parser/_parser.py:207: in split

    return list(cls(s))

_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <dateutil.parser._parser._timelex object at 0x7fad8244b1d0>

instream = datetime.datetime(2021, 2, 23, 19, 22, 59, 218277)

    def __init__(self, instream):

        if six.PY2:

            # In Python 2, we can't duck type properly because unicode has

            # a 'decode' function, and we'd be double-decoding

            if isinstance(instream, (bytes, bytearray)):

                instream = instream.decode()

        else:

            if getattr(instream, 'decode', None) is not None:

                instream = instream.decode()

    

        if isinstance(instream, text_type):

            instream = StringIO(instream)

        elif getattr(instream, 'read', None) is None:

            raise TypeError('Parser must be a string or character stream, not '

>                           '{itype}'.format(itype=instream.__class__.__name__))

E           TypeError: Parser must be a string or character stream, not datetime

.tox/py3/lib/python3.6/site-packages/dateutil/parser/_parser.py:76: TypeError

------------------------------ Captured log call -------------------------------

WARNING  connexion.operations.openapi3:openapi.py:265 this operation accepts multiple content types, using application/json

WARNING  connexion.operations.openapi3:openapi.py:265 this operation accepts multiple content types, using application/json

WARNING  connexion.operations.openapi3:openapi.py:265 this operation accepts multiple content types, using application/json

WARNING  connexion.operations.openapi3:openapi.py:265 this operation accepts multiple content types, using application/json

WARNING  connexion.operations.openapi3:openapi.py:265 this operation accepts multiple content types, using application/json

WARNING  connexion.operations.openapi3:openapi.py:265 this operation accepts multiple content types, using application/json

WARNING  connexion.operations.openapi3:openapi.py:265 this operation accepts multiple content types, using application/json

WARNING  connexion.operations.openapi3:openapi.py:265 this operation accepts multiple content types, using application/json
````

Reply:
This error is complaining that a datetime was passed in when a string or character stream must be passed in